Install/build error on MacOS 13.0.1 in Python 3.8.15 with pyarrow-10.0.1 but pyarrow-9.0.0 works

Got pages of the same error. 
```
In file included from src/turbodbc_arrow/arrow_result_set.cpp:6:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1745:36: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
    if (IsFieldPath()) return std::get<FieldPath>(impl_).indices().size() > 1;
                                   ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/turbodbc_arrow/arrow_result_set.cpp:6:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/Users/jianfengmao/Playground/venv3.8/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1750:34: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
    return IsFieldPath() ? &std::get<FieldPath>(impl_) : NULLPTR;
                                 ^
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -3,6 +3,11 @@ Version history / changelog
 
 From version 2.0.0, turbodbc adapts semantic versioning.
 
+Version 4.5.7
+-------------
+
+* Set ``_LIBCPP_DISABLE_AVAILABILITY`` to allow building with C++17 on macOS.
+
 Version 4.5.6
 -------------
 

--- a/setup.py
+++ b/setup.py
@@ -85,6 +85,8 @@ if sys.platform == "darwin":
     extra_compile_args.append("--std=c++17")
     extra_compile_args.append("--stdlib=libc++")
     extra_compile_args.append("-mmacosx-version-min=10.9")
+    # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+    extra_compile_args.append("-D_LIBCPP_DISABLE_AVAILABILITY")
     hidden_visibility_args.append("-fvisibility=hidden")
     include_dirs.append(os.getenv("UNIXODBC_INCLUDE_DIR", "/usr/local/include/"))
     library_dirs.append(os.getenv("UNIXODBC_LIBRARY_DIR", "/usr/local/lib/"))

